iree/compiler/Dialect/Flow/Transforms/test/deduplicate_executables.mlir

```diff
@@ -0,0 +1,141 @@
+      %0 = mhlo.subtract %arg0, %arg0 : tensor<4xf32>
+      return %0 : tensor<4xf32>
+    }
+  }
+}
+// CHECK-LABEL: func @multiple_entry_points
+func @multiple_entry_points(%arg0: tensor<4xf32>) -> tensor<4xf32> {
+  %c4 = constant 4 : index
+  // CHECK: %0 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_0[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  %0 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_0[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  // CHECK: %1 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_1[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  %1 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_1[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  // CHECK: %2 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_0[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  %2 = flow.dispatch @multiple_entry_points_ex_1::@multiple_entry_points_1_entry_0[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  // CHECK: %3 = flow.dispatch @multiple_entry_points_ex_0::@multiple_entry_points_0_entry_1[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  %3 = flow.dispatch @multiple_entry_points_ex_1::@multiple_entry_points_1_entry_1[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xf32>
+  return %0 : tensor<4xf32>
+}
+
+// -----
+
+// CHECK-LABEL: flow.executable @different_types_float_ex
+flow.executable @different_types_float_ex {
+  flow.dispatch.entry @different_types_float_entry
+  module {
+    func @different_types_float_entry(%arg0: tensor<4xf32>) -> tensor<4xi1> {
+      %0 = "mhlo.compare"(%arg0, %arg0) {comparison_direction = "EQ"} : (tensor<4xf32>, tensor<4xf32>) -> tensor<4xi1>
+      return %0 : tensor<4xi1>
+    }
+  }
+}
+// CHECK-LABEL: flow.executable @different_types_int_ex
+flow.executable @different_types_int_ex {
+  flow.dispatch.entry @different_types_int_entry
+  module {
+    func @different_types_int_entry(%arg0: tensor<4xi32>) -> tensor<4xi1> {
+      %0 = "mhlo.compare"(%arg0, %arg0) {comparison_direction = "EQ"} : (tensor<4xi32>, tensor<4xi32>) -> tensor<4xi1>
+      return %0 : tensor<4xi1>
+    }
+  }
+}
+// CHECK-LABEL: func @different_types
+func @different_types(%arg0: tensor<4xf32>) -> tensor<4xi1> {
+  %c4 = constant 4 : index
+  // CHECK: %0 = flow.dispatch @different_types_float_ex::@different_types_float_entry[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xi1>
+  %0 = flow.dispatch @different_types_float_ex::@different_types_float_entry[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xi1>
+  // CHECK: %1 = flow.dispatch @different_types_int_ex::@different_types_int_entry[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xi1>
+  %1 = flow.dispatch @different_types_int_ex::@different_types_int_entry[%c4 : index](%arg0) : (tensor<4xf32>) -> tensor<4xi1>
+  return %0 : tensor<4xi1>
+}
```

Comment:
add a test with a nested op like mhlo.reduce:
```
%2 = "mhlo.reduce"(%0, %1) ({
    ^bb0(%arg3: tensor<f32>, %arg4 : tensor<f32>):
      %3 = mhlo.add %arg3, %arg4 : tensor<f32>
      "mhlo.return"(%3) : (tensor<f32>) -> ()
    }) {dimensions = dense<1> : tensor<1xi64>} : (tensor<5x4xf32>, tensor<f32>) -> tensor<5xf32>
```

then you can verify that if the inner ops differ (add vs max or something) your equivalence check works